src/nvim/os/env.c

```diff
@@ -169,6 +178,64 @@ void os_get_hostname(char *hostname, size_t size)
 #endif
 }
 
+/// @return -1 if error encountered, 1 if capslock active otherwise 0 if not.
+int capslock_status(void ) {
+
+#if defined(__linux__)
+
+  FILE *fp = NULL;
+  char res[1024];
+  fp = popen("xset -q | grep Caps", "r");
+  if (fp == NULL) {
+      return -1;
+  }
+  fgets(res, sizeof(res), fp);
+  if (STRNCMP(res + 21, "on", 2) == 0) {
```

Comment:
This is making a lot of assumptions about the output from `xset`.  How portable are those assumptions?  Does xset provide a mechanism to request parseable output?  Is there another way to request this information?

Reply:
And do not use `STRNCMP` here, with *proper* (i.e. `char` and not `char_u`) types `strncmp` is good and it does type check in place of forcing a cast.

Reply:
OK, I will do it, I thought here would be no difference for char and unsigned char, will do it

Reply:
There is no difference after compiling, but macros would be forcing a cast.